wolfcrypt/src/logging.c

```diff
@@ -957,25 +1137,108 @@ int wc_ERR_remove_state(void)
         current = next;
     }
 
-    wc_error_queue_count = 0;
+    wc_errors_count = 0;
     wc_errors          = NULL;
     wc_last_node       = NULL;
 
-    wc_UnLockMutex(&debug_mutex);
+    ERRQ_UNLOCK();
 
     return 0;
 }
 
-#if !defined(NO_FILESYSTEM) && !defined(NO_STDIO_FILESYSTEM)
-/* empties out the error queue into the file */
-static int wc_ERR_dump_to_file (const char *str, size_t len, void *u)
+unsigned long wc_PeekErrorNodeLineData(const char **file, int *line,
+                                       const char **data, int *flags,
+                                       int (*ignore_err)(int err))
 {
-    XFILE fp = (XFILE ) u;
-    if (fprintf(fp, "%-*.*s\n", (int)len, (int)len, str) < 0)
-        return IO_FAILED_E;
-    return 0;
+    int ret = 0;
+    int idx;
+
+    WOLFSSL_ENTER("wc_PeekErrorNodeLineData");
+
+/* No data or flags stored - error display only in Nginx. */
+    if (data != NULL) {
+        *data = "";
+    }
+    if (flags != NULL) {
+        *flags = 0;
+    }
+
+    if (ERRQ_LOCK() != 0) {
+        WOLFSSL_MSG("Lock debug mutex failed");
+        return BAD_MUTEX_E;
+    }
+
+    idx = getErrorNodeCurrentIdx();
+    while (1) {
+        ret = peekErrorNode(idx, file, NULL, line);
+        if (ret == BAD_MUTEX_E || ret == BAD_FUNC_ARG || ret == BAD_STATE_E) {
+            WOLFSSL_MSG("Issue peeking at error node in queue");
+            return 0;
+        }
+        /* OpenSSL uses positive error codes */
+        if (ret < 0) {
+            ret = -ret;
+        }
+
+        if (ignore_err && ignore_err(ret)) {
+          removeErrorNode(idx);
+          continue;
+        }
+
+        ERRQ_UNLOCK();
+        return (unsigned long)ret;
+    }
+}
+
+unsigned long wc_GetErrorNodeErr(void)
```

Comment:
Even though the name is pretty clear on this function, it still could use a single comment line at the minimum to describe the return values (i.e. is it the error from head of list / tail of list, is the error pulled out of queue).

Reply:
Honestly, I am not 100% sure what this function does. The whole `current` handling of the previous implementation...I can only make a vague guess on how it is supposed to work.

I'll add comments to the thread-local variants.